### ISSUE TITLE
"ValueError: too many values to unpack" during 2.4.1 cluster creation

### ISSUE THREAD

**Issue by tgjohnst** [aws/aws-parallelcluster#1241]
I just updated to 2.4.1 and am now also experiencing the same ValueError during cluster creation as was referenced in #1236

**Environment:**
 - AWS ParallelCluster version 2.4.1
 - OS: ubuntu1604
 - Scheduler: sge
 - Master instance type: r5.2xlarge
 - Compute instance type: r5.2xlarge

**Bug description and how to reproduce:**
Using the same template that worked fine in 2.4.0 (the only change being using 2.4.1 AMI), I now receive an error when running *pcluster create*
```
$ pcluster create rosalind
Beginning cluster creation for cluster: rosalind
Unexpected error of type ValueError: too many values to unpack (expected 2)
```

See cluster config attached:
[pcluster_config_RCE.txt](https://github.com/aws/aws-parallelcluster/files/3448734/pcluster_config_RCE.txt)

Any idea why 2.4.1 is now throwing this error?

**Comment by demartinofra** [comment]
Hi,
thanks for reporting this. Unfortunately I was unable to reproduce the issue. Could you please install this version and rerun the failing command? [aws-parallelcluster-2.4.1.tar.gz](https://github.com/aws/aws-parallelcluster/files/3451736/aws-parallelcluster-2.4.1.tar.gz)

`pip install --upgrade aws-parallelcluster-2.4.1.tar.gz`

I simply removed the exception handling so that we should be able to see the stack trace on unexpected failures.

Thanks,
Francesco

**Comment by tgjohnst** [comment]
Thanks Francesco, here's the stack trace:
```Beginning cluster creation for cluster: rosalind
Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/bin/pcluster", line 10, in <module>
    sys.exit(main())
  File "/home/ubuntu/miniconda3/lib/python3.7/site-packages/pcluster/cli.py", line 371, in main
    args.func(args)
  File "/home/ubuntu/miniconda3/lib/python3.7/site-packages/pcluster/cli.py", line 29, in create
    pcluster.create(args)
  File "/home/ubuntu/miniconda3/lib/python3.7/site-packages/pcluster/pcluster.py", line 77, in create
    config = cfnconfig.ParallelClusterConfig(args)
  File "/home/ubuntu/miniconda3/lib/python3.7/site-packages/pcluster/cfnconfig.py", line 77, in __init__
    self.__init_cluster_parameters()
  File "/home/ubuntu/miniconda3/lib/python3.7/site-packages/pcluster/cfnconfig.py", line 526, in __init_cluster_parameters
    self.__validate_resource(cluster_options.get(key)[1], __temp__)
  File "/home/ubuntu/miniconda3/lib/python3.7/site-packages/pcluster/cfnconfig.py", line 289, in __validate_resource
    self.__resource_validator.validate(resource_type, resource_value)
  File "/home/ubuntu/miniconda3/lib/python3.7/site-packages/pcluster/config_sanity.py", line 354, in validate
    for actions, resource_arn in iam_policy:
ValueError: too many values to unpack (expected 2)```

**Comment by demartinofra** [comment]
Thanks, that really helped.

Unfortunately there is a bug in the sanity check when a custom ec2_iam_role is specified in the config. For the moment you could set `sanity_check = false` to unblock cluster creation.

Here is the fix: https://github.com/aws/aws-parallelcluster/pull/1246

**Comment by tgjohnst** [comment]
Thanks, disabling the sanity checks will work for now!

Still errored out as it looks like the IAM permission autoscaling:EnableMetricsCollection is now needed as part of 2.4.1 (didn't notice that in the release notes), but that's a separate issue. should be good to go after adding that to the policy.

**Comment by enrico-usai** [comment]
Hi @tgjohnst the EnableMetricsCollection has been added to the guide:
https://github.com/awsdocs/aws-parallelcluster-user-guide/pull/9/files

Thank you

**Comment by sean-smith** [comment]
Hi @tgjohnst 

Sorry, we missed this in the ChangeLog. I've added it https://github.com/aws/aws-parallelcluster/pull/1250
